imports/ui/components/dashboard/match_card.jsx

```diff
@@ -1,49 +1,123 @@
 import React, { Component } from 'react';
-import { Card, Dropdown, ButtonGroup, Button } from 'react-bootstrap';
+import { Card, Dropdown, ButtonGroup, Button, Modal, Form } from 'react-bootstrap';
 import '/imports/ui/styles/_cards.scss';
 import { PropTypes } from 'prop-types';
 
 class MatchCard extends Component {
   constructor(props) {
     super(props);
+    this.showFlagModal = this.showFlagModal.bind(this);
+    this.closeFlagModal = this.closeFlagModal.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.submitFlagModal = this.submitFlagModal.bind(this);
+    this.state = {
+      showFlagModal: false,
+      flagReason: '',
+    };
+  }
+  showFlagModal(event) {
+    event.preventDefault();
+    this.setState({ showFlagModal: true });
+  }
+  closeFlagModal(event) {
+    event.preventDefault();
+    this.setState({ showFlagModal: false });
+  }
+  handleChange(e) {
+    this.setState({
+      flagReason: e.target.value,
+    });
+  }
+  submitFlagModal(e) {
+    e.preventDefault();
+
+    const data = {
+      userId: Meteor.userId(),
+      reason: this.state.flagReason,
+      entryId: this.props.entryId,
+    };
+
+    this.setState({ showFlagModal: false });
+    Meteor.call('entry.flag', data, (error, result) => {
+      if (error) {
+        this.setState({ error: error.reason });
```

Comment:
can this be better written with passing a single state object instead of calling `setState()` twice?
```javascript
if (error) {
   this.setState({ error: error.reason, processing: false })
}
```

Reply:
Good call, thanks!